sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala

```diff
@@ -692,4 +692,18 @@ class JoinHintSuite extends PlanTest with SharedSparkSession with AdaptiveSparkP
     assert(logs.size == 2)
     logs.forall(_.contains("no equi-join keys"))
   }
+
+  test("SPARK-36652: AQE dynamic join selection should not apply to non-equi join") {
+    val hintAppender = new LogAppender(s"join hint check for equi-join")
+    withLogAppender(hintAppender, level = Some(Level.WARN)) {
+      withSQLConf(
+        SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
+        SQLConf.ADAPTIVE_MAX_SHUFFLE_HASH_JOIN_LOCAL_MAP_THRESHOLD.key -> "67108865") {
```

Comment:
what's this magic number?

Reply:
It's due to `advisoryPartitionSize <= maxShuffledHashJoinLocalMapThreshold` in https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/DynamicJoinSelection.scala#L48. `spark.sql.adaptive.advisoryPartitionSizeInBytes` by default is 64MB, which is 67108864. +1 one here to trigger the bug. This is just an example.

Reply:
I can change the number to `64MB` if it's preferred.

Reply:
yea '64MB' seems better

Reply:
@cloud-fan - sure, updated.